mm2src/mm2_bitcoin/chain/src/block_header.rs

```diff
@@ -235,7 +235,8 @@ impl Deserializable for BlockHeader {
             None
         };
 
-        let is_zcash = version == 4 && !reader.coin_variant().is_btc() && !reader.coin_variant().is_ppc();
+        let is_zcash = (version == 4 && !reader.coin_variant().is_btc_or_ppc())
+            || (version == 1 && reader.coin_variant().is_rick());
```

Comment:
No need to check for version 1 here, the below code is enough
```rust
let is_zcash = version == 4 && !reader.coin_variant().is_btc_or_ppc() || reader.coin_variant().is_rick();
```

Reply:
done

Reply:
done